src/mockFA2.ml

```diff
@@ -66,8 +70,10 @@ let[@inline] fa2_run_transfer (state, xs: fa2_state * fa2_transfer list) : fa2_s
       xs in
   (state, ([]: LigoOp.operation list))
 
-let[@inline] transfer (state: fa2_state) (xs: fa2_transfer list) : LigoOp.operation list * fa2_state =
-  let state, ops = fa2_run_transfer (state, xs) in
+let[@inline] transfer (state: mock_fa2_state) (xs: fa2_transfer list) : LigoOp.operation list * mock_fa2_state =
+  let { fa2_state = fa2_state; metadata = metadata; } = state in (* deconstruct *)
+  let fa2_state, ops = fa2_run_transfer (fa2_state, xs) in
+  let state = { fa2_state = fa2_state; metadata = metadata; } in (* reconstruct *)
```

Comment:
Minor point, but it might be good to keep the syntax uniform for handling the reconstruction of the final state (we use this approach here and `{st with fa2_state = fa2_state}` in `wctez.ml`). Not sure which is better, but perhaps the latter reduces the likelihood of accidentally messing with the metadata.

Reply:
Ah, nice point, yes. In general explicit deconstruction/reconstruction I think is more efficient (gas-wise).

The field update reduces the likelihood of accidentally messing with the metadata, true, but it's pretty hard to create bytes in our current setup so I wouldn't worry about that very much. 

On the other hand the field update will type-check without errors/warnings when we update the state type which can be dangerous (especially in the case of the tez wrapper, where sometimes we need to update the vaults and the fa2_state, but other times we only need to update the fa2_state). Given this, I'd uniformly switch to explicit deconstruction/reconstruction.

Reply:
Okay great, sounds good to me!

Reply:
Done in c5f239a!

I noticed that we still use field updates in `fa2_run_update_operators` (for all FA2 modules) but I am not gonna risk touching that part unnecessarily, since it does not relate to the changes in this PR (and, more importantly, I'd hate to introduce bugs now just to refactor this :sweat_smile:)